R/board_folder.R

```diff
@@ -71,16 +71,16 @@ pin_delete.pins_board_folder <- function(board, names, ...) {
 
 #' @export
 pin_store.pins_board_folder <- function(board, name, paths, metadata,
-                                              versioned = NULL, ...) {
+                                        versioned = NULL, ...) {
   check_pin_name(name)
   version <- version_setup(board, name, version_name(metadata), versioned = versioned)
-
-  version_dir <- fs::path(board$path, name, version)
-  fs::dir_create(version_dir)
-  write_meta(metadata, version_dir)
-  out_paths <- fs::file_copy(paths, version_dir, overwrite = TRUE)
-  fs::file_chmod(out_paths, "u=r")
-
+  if (!is.null(version)) {
+    version_dir <- fs::path(board$path, name, version)
+    fs::dir_create(version_dir)
+    write_meta(metadata, version_dir)
+    out_paths <- fs::file_copy(paths, version_dir, overwrite = TRUE)
+    fs::file_chmod(out_paths, "u=r")
+  }
```

Comment:
Step 2: Then here in `pin_store` when the version is NULL, we skip all the writing (do not store) and just return the `name`.